src/plot_api/plot_api.js

```diff
@@ -1624,6 +1624,11 @@ function _restyle(gd, aobj, _traces) {
                     flags.docalc = true;
                 }
 
+                // some attributes declare a 'recalc' flag
+                if(valObject.recalc) {
+                    flags.docalc = true;
```

Comment:
... slowly migrating to something better than those ugly `recalcAttrs` list in restyle and relayout.